src/rawposix/src/fs_calls.rs

```diff
@@ -1,14 +1,30 @@
 use libc::c_void;
+use std::ffi::CStr;
 use typemap::datatype_conversion::*;
 use typemap::path_conversion::*;
 use sysdefs::constants::err_const::{syscall_error, Errno, get_errno, handle_errno};
 use sysdefs::constants::fs_const::{STDIN_FILENO, STDOUT_FILENO, STDERR_FILENO, O_CLOEXEC, MAP_ANONYMOUS, MAP_FIXED, MAP_PRIVATE, MAP_SHARED, PROT_EXEC, PROT_NONE, PROT_READ, PROT_WRITE, PAGESHIFT, PAGESIZE};
-use sysdefs::constants::lind_platform_const::{FDKIND_KERNEL, MAXFD};
+use sysdefs::constants::lind_platform_const::{FDKIND_KERNEL, MAXFD, LIND_ROOT};
 use sysdefs::constants::sys_const::{DEFAULT_UID, DEFAULT_GID};
+use sysdefs::data::fs_struct::StatData;
 use typemap::cage_helpers::*;
 use cage::{round_up_page, get_cage, HEAP_ENTRY_INDEX, MemoryBackingType, VmmapOps};
 use fdtables;
 
+/// Panic wrapper for unused argument validation failures
```

Comment:
If this is helpful we should have this hosted in a different file and be used in other PRs. @Yaxuan-w thoughts?